website/blog/2022-10-18-how-to-design-metrics.md

```diff
@@ -0,0 +1,367 @@
+---
+title: "How to design metrics: deciding what to model in the modeling layer vs the semantic layer"
+description: "placeholder"
+slug: how-to-design-metrics
+
+authors: [callum_mccann]
+
+tags: [dbt product updates]
+hide_table_of_contents: false
+
+date: 2022-10-18
+is_featured: true
+---
+
+---
+
+**IMPORTANT:** This document serves as the temporary location for information on how to design and structure your metrics. It is our intention to take this content and turn it into a Guide, similar to [How we structure our dbt projects](https://docs.getdbt.com/guides/best-practices/how-we-structure/1-guide-overview) , but we feel that codifying information in a Guide first requires that Metrics be rigorously tested by the community so that best practices can arise. This document contains our early attempts to create best practices. In other words, read these as suggestions for a new paradigm and share in the community where they do (or don’t) match your experiences! More information on where to do this can be found at the end.
+
+---
+
+## The power of a semantic layer on top of a mature data modeling framework
+
+As a longtime dbt Community member, I knew I had to get involved when I first saw the dbt Semantic Layer. It gave me a vision of a world where metrics and business logic were unified across an entire organization; a world where the data team was no longer bound to a single consuming experience and could enable their stakeholders in dozens of different ways. To me, it felt like the opportunity to contribute to the next step of what dbt could become.
+
+In past roles, I’ve been referred to as the `dbt zealot` and I’ll own that title! But it’s not a surprise - dbt was built to serve analytics engineers exactly like me. It gave us flexibility and power to serve our organizations. But it always felt like dbt best served **us.**
+
+The Semantic Layer takes this to the next level **by coupling dbt’s mature data modeling framework with semantic definitions.** The result is a first of its kind data experience that serves both the analytics engineers and stakeholders. Metrics are the first step towards this vision, allowing users to version control and centrally define their key business metrics in a single repo **while also** serving them to the entire business.
+
+However, this is still a relatively new part of the dbt tool box and you probably have a lot of questions on how **exactly** you can do that. Contained within this blog are our early recommendations for how to design and structure metrics, along with examples so that you can apply the same principles to your use cases. They were developed by combining the overall philosophy of dbt with our hands on learning gathered during the beta period and internal testing.
```

Comment:
```suggestion
However, this is still a relatively new part of the dbt toolbox and you probably have a lot of questions on how **exactly** you can do that. This blog contains our early recommendations for how to design and structure metrics, along with examples so that you can apply the same principles to your use cases. We developed these recommendations by combining the overall philosophy of dbt, with our hands-on learning gathered during the beta period and internal testing.
```